src/App.tsx

```diff
@@ -7,33 +9,67 @@ function getRandomName(): string {
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+type State = {
+  hasClock: boolean,
+  clockName: string,
+};
+
+export class App extends React.Component<{}, State> {
+  state: Readonly<State> = {
+    hasClock: true,
+    clockName: 'Clock-0',
+  };
+
+  clockNameId = 0;
```

Comment:
A name like that can easily confuse those who read. We don't store `nameId` in this variable
```suggestion
  clockNameTimerId = 0;
```